filebeat/harvester/log.go

```diff
@@ -180,58 +210,42 @@ func (h *Harvester) openFile() error {
 		return err
 	}
 
-	// update file offset
-	err = h.initFileOffset(f)
+	// get file offset. Only update offset if no error
+	offset, err := h.getFileOffset(f)
 	if err != nil {
 		return err
 	}
 
-	// yay, open file
-	h.file = source.File{f}
+	logp.Debug("harvester", "Setting offset for file: %s. Offset: %d ", h.state.Source, offset)
+	h.state.Offset = offset
+
 	return nil
 }
 
-func (h *Harvester) initFileOffset(file *os.File) error {
-	offset, err := file.Seek(0, os.SEEK_CUR)
+func (h *Harvester) getFileOffset(file *os.File) (int64, error) {
```

Comment:
I think `initFileOffset` still matches this methods functionality better. `getFileOffset` sounds like re-usable getter.